share/mk/bsd.compat.mk

```diff
@@ -24,7 +26,7 @@ COMPAT_COMPILER_TYPE=${COMPILER_TYPE}
 
 # -------------------------------------------------------------------
 # 32 bit world
-.if ${COMPAT_ARCH} == "amd64"
+.if ${COMPAT_ARCH} == "amd64" && ${MK_LIB32} != "no"
```

Comment:
I think it might make more sense to wrap the whole chain of if/elif/endif's rather than duplicating the MK_LIB32 check.